web/src/actions/admin.test.js

```diff
@@ -474,4 +474,135 @@ describe('admin actions', () => {
       });
     });
   });
+
+  describe('edit own account', () => {
+    const user = { id: 7 };
+
+    it('handles a conflicting email address', async () => {
+      const store = mockStore({ notification: { open: false, queue: [] } });
+
+      fetchMock.onPut('/me').reply(400, { error: 'update-self-email-exists' });
+
+      try {
+        await store.dispatch(actions.editSelf(user));
+      } catch (_) {} // eslint-disable-line no-empty
+
+      expect(store.getActions()).toEqual(
+        expect.arrayContaining([
+          { type: actions.ADMIN_EDIT_ME_REQUEST },
+          { type: actions.ADMIN_EDIT_ME_ERROR },
+          {
+            type: ADD_NOTIFICATION,
+            message:
+              'Error: another account already exists with that email address'
+          }
+        ])
+      );
+      expect(store.getActions().length).toEqual(3);
+    });
+
+    it('handles an invalid phone number', async () => {
+      const store = mockStore({ notification: { open: false, queue: [] } });
+
+      fetchMock.onPut('/me').reply(400, { error: 'update-self-invalid-phone' });
+
+      try {
+        await store.dispatch(actions.editSelf(user));
+      } catch (_) {} // eslint-disable-line no-empty
```

Comment:
Since it's testing an error condition and the action returns a promise, it'll reject, so we want to catch that and then check the expectations.  We could do that inside the `catch` but that feels weird to me.

The `_` is commonly used in some languages to indicate a variable that you're not going to use.  Like if you have a function that takes two arguments but you only care about the first, you might declare it as `function (_, arg) { ... }`.  eslint also won't complain if you have an unused variable named `_`.  Here, Javascript requires you to catch *something* but since we're ignoring it, I made it `_`, which satisfies basically everything.  🙂